src/systems/buoyancy/Buoyancy.cc

```diff
@@ -114,6 +226,44 @@ void Buoyancy::Configure(const Entity &_entity,
   {
     this->dataPtr->fluidDensity = _sdf->Get<double>("uniform_fluid_density");
   }
+  else if(_sdf->HasElement("graded_buoyancy"))
+  {
+    this->dataPtr->buoyancyType =
+      BuoyancyPrivate::BuoyancyType::GRADED_BUOYANCY;
+
+    auto gradedElement = _sdf->GetFirstElement();
+    if(gradedElement == nullptr)
+    {
+      ignerr << "Unable to get element description" << std::endl;
+      return;
+    }
+
+    auto argument = gradedElement->GetFirstElement();
+    while(argument != nullptr)
+    {
+      if(argument->GetName() == "default_density")
```

Comment:
Since you're expecting only one of this element, you could get it with:

```
auto defaultDensityElem = gradedElement->GetElement("default_density");
```